pkg/resources/usercluster/deployment.go

```diff
@@ -374,3 +381,96 @@ func getLabelsArgValue(cluster *kubermaticv1.Cluster) (string, error) {
 	}
 	return string(bytes), nil
 }
+
+func getEnvVars(data userclusterControllerData) ([]corev1.EnvVar, error) {
```

Comment:
as this is duplicated in multiple places, could you extract it so that we need to maintain it in only one place?